include/Board.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "macros.h"
+#include "Piece.h"
+#include "Factory.h"
+
+#include <memory>
+#include <iostream>
+#include <vector>
+
+
+class Board {
+
+public:
+	Board(const std::string& bStr);
+	int codeRespone(const std::string& move);
+
+private:
+	bool m_currentPlayerIsWhite;
+	std::unique_ptr<Piece> m_board[ROWS][COLS];
+	Position m_king[2];
+	
+	void clearAndBuildBoard(const std::string& bStr);
+
+	bool isEmpty(const Position source) const;
+	bool isRival(const Position source) const;
+	bool isCurrentTurnPiece(const Position source) const;
+	bool isWayBloacked(const Position& source, const Position& dest) const;

```

Comment:
The Board class seems to have multiple responsibilities, Consider separating these responsibilities into different classes or functions to adhere to the Single Responsibility Principle.